### ISSUE TITLE
JSON tests failing

### ISSUE THREAD

**Issue by dcoomber** [topkecleon/telegram-bot-bash#154]
The JSON tests are failing on both `develop` and `master` branch.  Please set me straight if I'm doing something wrong.

```
$ dev/all-tests.sh
```

Output:
```
TEST: ./a-commit-test.sh
Running pre-commit hook
............................
Test all scripts with shellcheck
    OK
............................
Update Version of modified files
OK: ---- ./a-commit-test.sh
TEST: ./c-init-test.sh
Running init test ...
............................
   OK
Check new files after init ...
   OK
Test if JSON.sh/JSON.sh exists ...
Test Sourcing of bashbot.sh ...
Test Sourcing of commands.sh ...
   OK
Test bashbot.sh stat
OK: ---- ./c-init-test.sh
TEST: ./d-JSON.sh-test.sh
Running JSON.sh test ...
............................
Check JSON.sh ...
  ... JSON.sh -s -b -n
*** d-JSON.sh-test/JSON32034.log.result-1	2020-09-27 20:47:24.290000000 +0200
--- d-JSON.sh-test/JSON32034.log.in.out-1	2020-09-27 20:47:47.850000000 +0200
***************
*** 15,18 ****
  ["result",0,"message","date"]	1555934204
  ["result",0,"message","contact","phone_number"]	"222222"
  ["result",0,"message","contact","first_name"]	"ADAC Pannenhilfe"
! ["result",0,"message","contact","vcard"]	"BEGIN:VCARD\nVERSION:2.1\nN:Pannenhilfe;ADAC;;;\nFN:ADAC Pannenhilfe\nTEL;CELL;PREF:+49179222222\nTEL;X-Mobil:222222\nEND:VCARD"
--- 15,24 ----
  ["result",0,"message","date"]	1555934204
  ["result",0,"message","contact","phone_number"]	"222222"
  ["result",0,"message","contact","first_name"]	"ADAC Pannenhilfe"
! ["result",0,"message","contact","vcard"]	"BEGIN:VCARD
! VERSION:2.1
! N:Pannenhilfe;ADAC;;;
! FN:ADAC Pannenhilfe
! TEL;CELL;PREF:+49179222222
! TEL;X-Mobil:222222
! END:VCARD"
FAIL: ./d-JSON.sh-test.sh 0
FAILURE 2 / 3

Logfiles from run are in /tmp/bashbot.test8126
Do not forget to delete bashbot test files in /tmp!!
```

**Comment by gnadelwartz** [comment]
this seems to related to handling/converting of newlines, but does not happen on my Box.
i'll investigate, thanks for pointing out.

what OS and bash Version are you using?

**Comment by dcoomber** [comment]
I'm using Ubuntu 20.04 LTS running on WSL2 (Windows Subsystem for Linux).  My bash version is `GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)`.

I'll investigate further after work today as it might just be a setting I need to change on WSL.  Please let me know if you come across anything during your investigation.

**Comment by gnadelwartz** [comment]
great, thats a perfect reason to setup WSL as I wanted to explore it anyway :-)

**Comment by dcoomber** [comment]
I am experiencing this result on my Ubuntu 20.04 box too.

Bash:
```
$ bash --version
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
```

OS:
```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by gnadelwartz** [comment]
i think i forget to push a change fo the test data ... i'll have a look

**Comment by gnadelwartz** [comment]
hm, the test dataset looks ok and even a new repo clone is working on my box, looks like I have to setup a virtual ubuntu and see what happens.

**Comment by gnadelwartz** [comment]
OK, found the problem, "JSON.sh -s" behave different between suse and ubuntu.whats needed ist removal of -s option.
I have done this for the tests, but have to check if everthings works the same if I do for all JSON.sh calls ...

This may also the source of problems with multiline input found by an other one

**Comment by gnadelwartz** [comment]
https://github.com/topkecleon/telegram-bot-bash/commit/2a5d47df8d6bdfc3ac176a734bff39e3e4423b13

**Comment by dcoomber** [comment]
Thank you, @gnadelwartz .  I have merged changes into my fork and confirm this issue is resolved on Ubuntu 20.04 (including when running via WSL).

Excellent work!
